src/main/java/sirius/web/security/UserContext.java

```diff
@@ -218,7 +218,12 @@ private void bindUserToRequest(WebContext ctx) {
         }
 
         if (currentUser != null) {
-            return Optional.of(currentUser);
+            if (currentUser.isLoggedIn()) {
+                setCurrentUser(currentUser);
```

Comment:
I'd say this is redundant and can be skipped... (we're already reading currentUser)...